finality-aleph/src/party.rs

```diff
@@ -113,18 +120,23 @@ where
     error!(target: "afa", "Consensus party has finished unexpectedly.");
 }
 
-fn get_node_index(authorities: &[AuthorityId], my_id: &AuthorityId) -> Option<NodeIndex> {
+fn get_node_index(authorities: &[AuthorityId], keystore: &SyncCryptoStorePtr) -> Option<NodeIndex> {
+    let our_consensus_keys: HashSet<_> = SyncCryptoStore::keys(&**keystore, KEY_TYPE)
+        .unwrap()
+        .into_iter()
+        .collect();
+    trace!(target: "afa", "Found {:?} consensus keys in our local keystore {:?}", our_consensus_keys.len(), our_consensus_keys);
     authorities
         .iter()
-        .position(|a| a == my_id)
+        .position(|pkey| our_consensus_keys.contains(&pkey.into()))
```

Comment:
I wanted to suggest using [`SyncCryptoStorePtr::has_keys`](https://docs.rs/sp-keystore/0.9.0/sp_keystore/trait.SyncCryptoStore.html#tymethod.has_keys) method, but its interface is quite ugly, so nevermind...

Reply:
Yeah, another point is also that this way I can make a call to the keystore only once and just create a set, while using `has_keys` I would need to make a call for every authority, I guess.